lib/matplotlib/axes/_axes.py

```diff
@@ -5718,16 +5723,19 @@ def imshow(self, X, cmap=None, norm=None, *, aspect=None,
         `~matplotlib.pyplot.imshow` expects RGB images adopting the straight
         (unassociated) alpha representation.
         """
-        if aspect is None:
-            aspect = mpl.rcParams['image.aspect']
-        self.set_aspect(aspect)
         im = mimage.AxesImage(self, cmap=cmap, norm=norm,
                               interpolation=interpolation, origin=origin,
                               extent=extent, filternorm=filternorm,
                               filterrad=filterrad, resample=resample,
                               interpolation_stage=interpolation_stage,
                               **kwargs)
 
+        if (not im.is_transform_set()
+                or im.get_transform().contains_branch(self.transData)):
+            if aspect is None:
+                aspect = mpl.rcParams['image.aspect']
+            self.set_aspect(aspect)
```

Comment:
I agree ignoring explicit user input is bad, I think we should only skip doing something with aspect if `aspect` is None.

Reply:
I could also see a case for raising if both `transform` and `aspect` are passed...